backend/inspirehep/matcher/validators.py

```diff
@@ -0,0 +1,16 @@
+from inspire_utils.record import get_value
+
+
+def authors_validator(author, result):
+    record_author_identifiers = author.get("ids")
+    result_author_identifiers = get_value(result, "ids")
+    for author_id in record_author_identifiers:
+        result_identifiers = {
+            author_identifier["value"]
+            for author_identifier in result_author_identifiers
+            if author_identifier["schema"] == author_id["schema"]
+        }
+        author_identifiers = {author_id["value"]}
```

Comment:
this is works but is a bit clumsy. A better way would be to build two `set`s of `(schema, value)`s, then check that the set intersection is not empty.